src/libraries/Common/tests/System/Net/Http/ByteAtATimeContent.cs

```diff
@@ -33,7 +33,11 @@ protected override async Task SerializeToStreamAsync(Stream stream, TransportCon
             for (int i = 0; i < _length; i++)
             {
                 buffer[0] = (byte)i;
+#if !NETFRAMEWORK
                 await stream.WriteAsync(buffer);
+#else
+                await stream.WriteAsync(buffer, 0, buffer.Length);
+#endif
```

Comment:
Can we use extension methods here and elsewhere to avoid writing `#if` everywhere?

Reply:
These exist in a number of places (websockets and pipelines)

Reply:
The parameter's type is different. On .Net Core WriteAsync takes ReadOnlyMemory whereas on Framework it's an array. Do you suggest to add extension method like WriteAsync(ReadOnlyMemeory<byte>, start, length)?

Reply:
Something like:

```c#
static ValueTask WriteAsync(this Stream stream, ReadOnlyMemory<byte> memory)
{
    bool isArray = MemoryMarshal.TryGetArray(memory, out ArraySegment<byte> segment);
    Assert.True(isArray);

    return new ValueTask(stream.WriteAsync(segment.Array, segment.Offset, segment.Count));
}
```

Reply:
https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamExtensions.netstandard.cs